openedx/core/djangoapps/user_api/management/commands/populate_enterprise_learner_current_job.py

```diff
@@ -0,0 +1,150 @@
+""" Management command for populaing current job of learners in learner profile. """
+
+import logging
+
+from edx_rest_api_client.client import OAuthAPIClient
+from urllib.parse import urljoin
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from common.djangoapps.student.models import User, UserProfile
+from openedx.core.djangoapps.user_api import errors
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Command to populate current job of learners in extended learner profile.
+
+    This command will fetch the current job of learners from course-discovery service
+    and populate it in the extended learner profile in edx-platform. This command is
+    supposed to be run only once in the system.
+
+    Example usage:
+        $ # Update the current job of learners.
+        $ ./manage.py lms populate_enterprise_learner_current_job
+    """
+
+    help = 'Populates current job of learners in extended learner profile.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--oauth-host', required=False,
+                            default=settings.ENTERPRISE_BACKEND_SERVICE_EDX_OAUTH2_PROVIDER_URL)
+        parser.add_argument('--client-id', required=False,
+                            default=settings.ENTERPRISE_BACKEND_SERVICE_EDX_OAUTH2_KEY)
+        parser.add_argument('--client-secret', required=False,
+                            default=settings.ENTERPRISE_BACKEND_SERVICE_EDX_OAUTH2_SECRET)
+
+    def _get_edx_api_client(self, config):
+        return OAuthAPIClient(
+            base_url=config['oauth_host'],
+            client_id=config['client_id'],
+            client_secret=config['client_secret'],
+        )
+
+    def _fetch_learner_job_data(self, config):
+        """
+        Get the username and current job of learners from discovery service.
+
+        Returns:
+            list: List of dictionaries containing username and current job of learners.
+        """
+        try:
+            client = self._get_edx_api_client(config)
+
+            course_discovery_url = settings.COURSE_CATALOG_URL_ROOT
+            learner_jobs_endpoint = '/taxonomy/api/v1/learners-current-job/'
+            url = urljoin(course_discovery_url, learner_jobs_endpoint)
+
+            response = client.get(url)
+            response.raise_for_status()
+
+            return response.json() if response else []
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception('Error while fetching username and current_job data.')
```

Comment:
I am not sure if this log will serve any purpose? We are re-raising the exception below and it will print the whole stack trace anyway.

Reply:
On a second thought, do we really need `try...except` here? `response.raise_for_status()` will raise the exception if there are any errors with the API call. Also there is no retry mechanism here so I think we may not need the `try...except`